src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.cs

```diff
@@ -1111,7 +1111,12 @@ private void UpdateHyperlinks()
 				return;
 			}
 
-			var previousHyperLinks = _hyperlinks.Select(h => h.hyperlink).ToList();
+			var previousHyperLinks = new List<Hyperlink>(_hyperlinks.Count);
+			foreach (var hyperlink in _hyperlinks)
+			{
+				previousHyperLinks.Add(hyperlink.hyperlink);
+			}
+
```

Comment:
Great catch!

You should be able to use [`SelectToList`](https://github.com/unoplatform/uno/blob/a56adb4acbe45ec55cd6411a0c4a02eecf5527b3/src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions.Collections/CollectionExtensions.cs#L161) which does the same.

Reply:
@KevinFerati Thanks for the PR! Note you need to digi-sign the CLA agreement before your PR goes through. See CLA assistant note please. Thanks again! OSS FTW!

Reply:
@jeromelaban I think it will be updated with this then:
```suggestion
			var previousHyperLinks = _hyperlinks.Select(hyperlink => hyperlink.hyperlink).ToList();

```

Reply:
> Great catch!
> 
> You should be able to use [`SelectToList`](https://github.com/unoplatform/uno/blob/a56adb4acbe45ec55cd6411a0c4a02eecf5527b3/src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions.Collections/CollectionExtensions.cs#L161) which does the same.

Yes, you're right. Didn't know about these extensions. Thanks for the advice !

Reply:
The suggested changes have been applied in the same commit and the CLA issue has been resolved, so i'm resolving this conversation.

Reply:
Thank you for the adjustments @KevinFerati, appreciated!